src/lib.rs

```diff
@@ -652,14 +652,61 @@ impl Config {
         self.statik.unwrap_or_else(|| self.infer_static(name))
     }
 
+    fn pkg_config_from_path(&self) -> OsString {
+        if cfg!(target_os = "windows") {
+            // Resolve pkg-config in PATH to find the absolute path to pkg-config that we should
+            // use, so that we can skip Strawberry Perl's pure-perl implementation of pkg-config,
+            // which (despite its presence in PATH) is for internal use and not meant to be used as
+            // a MinGW distribution.
+            let pathexts = env::var_os("PATHEXT").map_or_else(
+                || Vec::from(["COM", "EXE", "BAT"].map(|v| OsStr::new(v).to_owned())),
```

Comment:
```suggestion
                || ["COM", "EXE", "BAT"].iter().map(|v| OsStr::new(v).to_owned()).collect::<Vec<_>>(),
```

or so, so it still works with Rust 1.30